Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+    end
+
+    local normalizedName = self:normalizedName(name);
+    if (not self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        return;
+    end
+
+    local plusOne = self.MaterializedData.DetailsByPlayerName[normalizedName].total;
+    plusOne = self:toPlusOne(plusOne + change);
+    self:setPlusOnes(normalizedName, plusOne);
+end
+
+--- Import a CSV or TSV data string
+---
+---@param data string
+---@param openOverview boolean (optional, default: false)
+---@param MetaData table (optional, default: auto generate new metadata)
+---@return boolean
+function PlusOnes:import(data, openOverview, MetaData)
+    GL:debug("PlusOnes:import");
+    -- Make sure all the required properties are available and of the correct type
+    if (GL:empty(data)) then
+        GL.Interface:get("PlusOnes.Importer", "Label.StatusMessage"):SetText("Invalid plus one data provided");
+        return false;
+    end
+
+    local Totals = {};
+
+    -- If the user copy/pasted from google sheets there will be addition quotes that need to be removed
+    data = data:gsub("\"", "");
+
+    for line in data:gmatch("[^\n]+") do
+        --- We expect lines to be in the form,
+        --- where the individual variables are comma, tab, or spaces separated:
+        --- PlayerName,Points
+        --- This facilitates imports from CSV, TSV files or pasting from Google Docs
+        local Segments = GL:separateValues(line);
+        
+        local playerName = tostring(Segments[1]);
+        playerName = GL:normalizedName(playerName);
+        local plusOne = self:toPlusOne(Segments[2]);
+
+        if (not GL:empty(playerName) and not Totals[playerName] and plusOne) then
+            Totals[playerName] = plusOne;
+        end
+    end
+
+    if (GL:empty(Totals)) then
+        local errorMessage = "Invalid data provided. Make sure that the contents follows the required format and no header row is included.";
+        GL.Interface:get("PlusOnes.Importer", "Label.StatusMessage"):SetText(errorMessage);
+
+        return false;
+    end
+
+    local MetaData = MetaData or {};
+
+    DB.PlusOnes = {
+        Totals = Totals,
+        MetaData = {
+            importedAt = MetaData.importedAt or GetServerTime(),
+            updatedAt = MetaData.updatedAt or GetServerTime(),
+            uuid = MetaData.uuid or GL:uuid(),
+        },
+    };
+
+    GL:success("Import of plus one data successful");
+    GL.Events:fire("GL.PLUSONES_IMPORTED");
+
+    self:materializeData();
+    GL.Interface.PlusOnes.Importer:close();
+
+    if (openOverview) then
+        self:draw();
+
+        -- The user is in charge of automatically sharing TMB data
+        -- after importing it, let's get crackin'!
+        if (GL.Settings:get("PlusOnes.automaticallyShareData")
+            and self:userIsAllowedToBroadcast()
+        ) then
+            self:broadcast();
+        end
+    end
+
+    return true;
+end
+
+--- Export to CSV
+---
+---@param displayFrame boolean
+---@return void
```

Comment:
Return type incorrect

Reply:
I used the same return type you used in Classes\BoostedRolls.lua